defensics/coap_automation_handler.py

```diff
@@ -87,18 +87,22 @@ def process(self, job):
     def run(self):
         self.tcp_server.start()
         self.proxy.run()
-
-        for data in self.tcp_server.recv():
+        received_data = self.tcp_server.recv()
+        for data in received_data:
             if self.proxy.is_ready():
                 try:
                     rsp = self.proxy.send(data)
                     self.tcp_server.send(rsp)
                 except TimeoutError:
                     logging.debug("Response timeout!")
+
                     self.proxy.device_iface.Disconnect()
-                    self.proxy.device_iface.Connect()
                     logging.debug("Disconnected")
-                    return 0
+                    self.proxy.device_iface.Connect()
+                    logging.debug("Reconnected")
```

Comment:
I don't think we know that we are indeed "reconnected" here. Maybe print something like "reconnecting"